akka-cluster/src/main/scala/akka/cluster/Cluster.scala

```diff
@@ -468,57 +469,101 @@ class Cluster(val system: ExtendedActorSystem) extends Extension {
   /**
    * INTERNAL API
    */
-  private[cluster] class ClusterLogger(log: LoggingAdapter) {
+  private[cluster] class ClusterLogger(log: MarkerLoggingAdapter) {
     def isDebugEnabled: Boolean =
       log.isDebugEnabled
 
     def logDebug(message: String): Unit =
-      logAtLevel(Logging.DebugLevel, message)
+      if (LogInfo && log.isDebugEnabled)
+        logAtLevel(Logging.DebugLevel, message)
 
     def logDebug(template: String, arg1: Any): Unit =
-      logAtLevel(Logging.DebugLevel, template, arg1)
+      if (LogInfo && log.isDebugEnabled)
```

Comment:
This looks a bit strange, perhaps we should spell it out a bit more to make clear `LogInfo` is `settings.LogInfo`?

Reply:
fixed